File: csharp-projects/Leetcode/0896-monotonic-array/0896-monotonic-array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isMonotonic = function(nums) {
    if(nums[0] < nums[nums.length - 1]) {
        for(let i=0; i<nums.length-1; i++) {
            if(nums[i] > nums[i+1]) return false;
        }
    }
    else if(nums[0] > nums[nums.length - 1]) {
        for(let i=0; i<nums.length-1; i++) {
            if(nums[i] < nums[i+1]) return false;
        }
    }
    else {
        for(let i=0; i<nums.length-1; i++) {
            if(nums[i] !== nums[i+1]) return false;
        }
    }
    return true;
};

File: csharp-projects/Leetcode/0896-monotonic-array/0896-monotonic-array.py
class Solution(object):
    def isMonotonic(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        if nums[0] < nums[-1]:
            for i in range(len(nums)-1):
                if nums[i] > nums[i+1]:
                    return False
        elif nums[0] > nums[-1]:
            for i in range(len(nums)-1):
                if nums[i] < nums[i+1]:
                    return False
        else:
            for i in range(len(nums)-1):
                if nums[i] != nums[i+1]:
                    return False
        return True